From eb6370987bb17e75c8285a104fe4b5ae60714eb0 Mon Sep 17 00:00:00 2001
From: KunYi <kunyi.chen@gmail.com>
Date: Mon, 2 Nov 2020 09:35:55 +0800
Subject: [PATCH] u-boot: add libubootenv

Add the new library libubootenv and remove fw_printenv and fw_setenv
form u-boot-tools as the are now part of the new library.

libubootenv is a library that provides a hardware independent
way to access to U-Boot environment. U-Boot has its default environment
compiled board-dependently and this means that tools to access the environment
are also board specific, too.

libubootenv conflicts with u-boot-tools from Debian 10
as both try to install fw_printenv and fw_sentenv. This conflict is not
part of the control file as it breaks the installation of custom u-boot-tools
from the u-boot-sources.
---
 .../libubootenv/libubootenv_0.3.bb            | 25 +++++++++++++++++++
 .../files/debian/u-boot-tools.conffiles       |  1 -
 .../u-boot/files/debian/u-boot-tools.install  |  2 --
 .../u-boot/files/debian/u-boot-tools.links    |  1 -
 4 files changed, 25 insertions(+), 4 deletions(-)
 create mode 100644 meta/recipes-bsp/libubootenv/libubootenv_0.3.bb
 delete mode 100644 meta/recipes-bsp/u-boot/files/debian/u-boot-tools.conffiles
 delete mode 100644 meta/recipes-bsp/u-boot/files/debian/u-boot-tools.links

diff --git a/meta/recipes-bsp/libubootenv/libubootenv_0.3.bb b/meta/recipes-bsp/libubootenv/libubootenv_0.3.bb
new file mode 100644
index 0000000..a0feed8
--- /dev/null
+++ b/meta/recipes-bsp/libubootenv/libubootenv_0.3.bb
@@ -0,0 +1,25 @@
+# libubootenv
+#
+# This software is a part of ISAR.
+# Copyright (c) Siemens AG, 2020
+#
+# SPDX-License-Identifier: MIT
+
+DESCRIPTION = "swupdate utility for software updates"
+HOMEPAGE= "https://github.com/sbabic/swupdate"
+LICENSE = "GPL-2.0"
+LIC_FILES_CHKSUM = "file://${LAYERDIR_isar}/licenses/COPYING.GPLv2;md5=751419260aa954499f7abaabaa882bbe"
+
+PROVIDES+="libubootenv0.1 libubootenv-dev libubootenv-tool"
+
+inherit dpkg-gbp
+
+SRC_URI = "git://salsa.debian.org/debian/libubootenv.git;protocol=https"
+
+SRCREV = "918da2525ce711700633ad69ea9b7e569b7abdbc"
+
+S = "${WORKDIR}/git"
+
+dpkg_runbuild_prepend() {
+	export DEB_BUILD_OPTIONS="nocheck"
+}
diff --git a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.conffiles b/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.conffiles
deleted file mode 100644
index d49a8fb..0000000
--- a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.conffiles
+++ /dev/null
@@ -1 +0,0 @@
-/etc/fw_env.config
diff --git a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.install b/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.install
index d1ae3e0..2893b9a 100644
--- a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.install
+++ b/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.install
@@ -1,5 +1,3 @@
 tools/dumpimage		/usr/bin/
-tools/env/fw_printenv	/usr/bin/
 tools/mkenvimage	/usr/bin/
 tools/mkimage		/usr/bin/
-tools/env/fw_env.config	/etc
diff --git a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.links b/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.links
deleted file mode 100644
index 92f5a6c..0000000
--- a/meta/recipes-bsp/u-boot/files/debian/u-boot-tools.links
+++ /dev/null
@@ -1 +0,0 @@
-/usr/bin/fw_printenv /usr/bin/fw_setenv
-- 
2.17.1

